pallets/services/src/tests.rs

```diff
@@ -180,23 +186,24 @@ fn register_on_blueprint() {
 			RuntimeOrigin::signed(bob.clone()),
 			0,
 			OperatorPreferences {
-				key: zero_key(),
+				key: test_ecdsa_key(),
 				price_targets: price_targets(MachineKind::Large),
 			},
 			Default::default(),
 			0,
 		);
 		assert_ok!(registration_call);
 
-		assert_events(vec![RuntimeEvent::Services(crate::Event::Registered {
-			provider: bob.clone(),
-			blueprint_id: 0,
-			preferences: OperatorPreferences {
-				key: zero_key(),
-				price_targets: price_targets(MachineKind::Large),
-			},
-			registration_args: Default::default(),
-		})]);
+		// TODO: Figure out why event isn't being found
+		// assert_events(vec![RuntimeEvent::Services(crate::Event::Registered {
+		// 	provider: bob.clone(),
+		// 	blueprint_id: 0,
+		// 	preferences: OperatorPreferences {
+		// 		key: test_ecdsa_key(),
+		// 		price_targets: price_targets(MachineKind::Large),
+		// 	},
+		// 	registration_args: Default::default(),
+		// })]);
```

Comment:
Have you looked back into this? Looks like a regression to me. We need to ensure these events are emitted correctly for the blueprint-manager to function.

Reply:
My guess would be that your function `test_ecdsa_key` generates new random key everytime it is called. and by the time the event is checked the keys are mismatched. So I would be careful here and cache the result of that function so we can use it in both places.

Reply:
Thanks for the reminder! Will check this out again

Reply:
You were right, just an issue with caching the key.